cdap-data-fabric/src/main/java/co/cask/cdap/data2/dataset2/lib/partitioned/PartitionedFileSetDataset.java

```diff
@@ -591,6 +595,14 @@ public PartitionOutput getPartitionOutput(PartitionKey key) {
   byte[] assertNotExists(PartitionKey key, boolean supportNonTransactional) {
     byte[] rowKey = generateRowKey(key, partitioning);
     if (tx == null && supportNonTransactional) {
+      if (LOG.isWarnEnabled()) {
+        StringBuilder sb = new StringBuilder();
+        for (StackTraceElement stackTraceElement : Thread.currentThread().getStackTrace()) {
```

Comment:
I think including the stacktrace in the log is a good way of letting the user which part of the code we're referring to (when we say it should be done in a transaction), since a single LOG.warn may not be enough to pinpoint them to the location in their code.